tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/unreachability_test.bal

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function testUnreachableCodeWithWhile1() {
+    while false {
+        int a = 1; // unreachable code
+        string b = "ABC";
+    }
+    int|string c = 25;
+}
+
+function testUnreachableCodeWithWhile2() {
+    while false {
+        error:unreachable();
+        foo();
+        string b = "ABC";
+        string c = b + "D";
+    }
+    int|string c = 25;
+}
+
+function testUnreachableCodeWithIf1() {
+    if false { // unreachable code
+        int a = 1;
+        string b = "ABC";
+    }
+    int|string c = 25;
+}
+
+function testUnreachableCodeWithIf2() {
+    if false { // unreachable code
+        error:unreachable();
+        string b = "ABC";
+        foo();
+        int a = 1;
+    }
+    int|string c = 25;
+}
+
+function testUnreachableCodeWithIf3() {
+    if true {
```

Comment:
`if true && true` cases seems not handled

Reply:
```ballerina
    if true {
	int a = 2;
    } else if true { // unreachable code
        int a = 1;
    }
```
This also compiles without an error.